fix(cordump_thread): get coredumpctl output in json format

fruch opened the pull request:
Switch to used json format when available, to avoid
parsing the outout of coredumpctl which keep of changing

Implementation of #5492 

(cherry picked from commit 5a3b46bad5f9fcdd07fb6a496360b835a8031e4c)

## PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

fgelcer commented on sdcm/utils/version_utils.py:
```diff
@@ -71,3 +77,12 @@ def is_enterprise(version):
+            return int(match.group("version"))
+        except ValueError:
+            pass
+    return 0
```
missing an empty line in the end of this file

fruch replied:
don't care..., this branch is breaking all of the checks we have anyhow..

fgelcer reviewed (approved):
other than a nitpick comment, LGTM

roydahan commented:
@fruch I reverted it from branch-perf-v8 because it broke it.
see here:
https://jenkins.scylladb.com/job/scylla-master/job/scylla-master-perf-regression-aws-test-throughput/300/console

fruch commented:
> @fruch I reverted it from branch-perf-v8 because it broke it.
> see here:
> https://jenkins.scylladb.com/job/scylla-master/job/scylla-master-perf-regression-aws-test-throughput/300/console

Yeah, old python  version that doesn't have the walrus operator

fruch commented:
> > @fruch I reverted it from branch-perf-v8 because it broke it.
> > see here:
> > https://jenkins.scylladb.com/job/scylla-master/job/scylla-master-perf-regression-aws-test-throughput/300/console
> 
> Yeah, old python version that doesn't have the walrus operator

fixed it in https://github.com/scylladb/scylla-cluster-tests/pull/5501
